pkg/steps/template.go

```diff
@@ -382,21 +382,23 @@ func isPodCompleted(podClient coreclientset.PodInterface, name string) (bool, er
 }
 
 func waitForTemplateInstanceReady(templateClient templateclientset.TemplateInstanceInterface, instance *templateapi.TemplateInstance) (*templateapi.TemplateInstance, error) {
-	for {
-		ready, err := templateInstanceReady(instance)
-		if err != nil {
-			return nil, fmt.Errorf("could not determine if template instance was ready: %v", err)
+	var actualErr error
+	err := wait.PollImmediate(2*time.Second, 10*time.Minute, func() (bool, error) {
+		instance, actualErr = templateClient.Get(instance.Name, meta.GetOptions{})
+		if actualErr != nil {
+			return false, nil
```

Comment:
We never fall out on perma-failing on this get anymore, except for the timeout. That's probably fine?

Reply:
I thought it was and in the spirit of retrying as long as possible

Reply:
and before you were looping forever if the ready check did not return an error but ready was always false